tests/amor/utils_test.py

```diff
@@ -0,0 +1,248 @@
+    with pytest.raises(ValueError, match='must have wavelength and theta coord'):
+        wavelength_theta_figure(da.hist(wavelength=3))
+
+    with pytest.raises(ValueError, match='must have wavelength and theta coord'):
+        wavelength_theta_figure(da.hist(theta=3))
+
+    assert wavelength_theta_figure(da.hist(wavelength=3, theta=3))
+
+
+def test_wavelength_figure_multiple_datasets(da, wavelength_bins, theta_bins):
+    assert wavelength_theta_figure(
+        (
+            da.bin(wavelength=10, theta=10),
+            da.hist(wavelength=10, theta=10),
+        ),
+    )
+    assert wavelength_theta_figure(
+        (
+            da,
+            da.bin(wavelength=10, theta=10),
+            da.hist(wavelength=10, theta=10),
+        ),
+        wavelength_bins=wavelength_bins,
+        theta_bins=theta_bins,
+    )
+    assert wavelength_theta_figure(
+        (da, da),
+        wavelength_bins=(wavelength_bins, wavelength_bins),
+        theta_bins=theta_bins,
+    )
+
+    with pytest.raises(ValueError, match=r'zip\(\) argument .*'):
+        assert wavelength_theta_figure(
+            (da, da),
+            # If sequence of bins is supplied, must have right length
+            wavelength_bins=(wavelength_bins,),
+            theta_bins=theta_bins,
+        )
+
+    assert wavelength_theta_figure(
+        da,
+        theta_bins=theta_bins,
+        wavelength_bins=wavelength_bins,
+        q_edges_to_display=(sc.scalar(1.0, unit='1/angstrom'),),
+    )
+    assert wavelength_theta_figure(
+        da,
+        theta_bins=theta_bins,
+        wavelength_bins=wavelength_bins,
+        # Can pass plot kwargs
+        grid=True,
+    )
+
+
+def test_q_figure_table(da, q_bins, theta_bins):
+    with pytest.raises(ValueError, match='binning provided'):
+        q_theta_figure(da)
+
+    with pytest.raises(ValueError, match='binning provided'):
+        q_theta_figure(da, q_bins=q_bins)
+
+    with pytest.raises(ValueError, match='binning provided'):
+        q_theta_figure(da, theta_bins=theta_bins)
+
+    assert q_theta_figure(da, theta_bins=theta_bins, q_bins=q_bins)
+
+
+def test_q_figure_binned(da, q_bins, theta_bins):
+    with pytest.raises(ValueError, match='binning provided'):
+        q_theta_figure(da.bin(Q=3))
+
+    with pytest.raises(ValueError, match='binning provided'):
+        q_theta_figure(da.bin(theta=3))
+
+    assert q_theta_figure(da.bin(Q=3, theta=3))
+    assert q_theta_figure(da.bin(Q=3), theta_bins=theta_bins)
+    assert q_theta_figure(da.bin(theta=3), q_bins=q_bins)
+
+
+def test_q_figure_hist(da, q_bins, theta_bins):
+    with pytest.raises(ValueError, match='must have theta and Q coord'):
+        q_theta_figure(da.hist(Q=3))
+
+    with pytest.raises(ValueError, match='must have theta and Q coord'):
+        q_theta_figure(da.hist(theta=3))
+
+    assert q_theta_figure(da.hist(Q=3, theta=3))
+
+
+def test_q_figure_multiple_datasets(da, q_bins, theta_bins):
```

Comment:
Following what we read in the google software book, it feels that these should all be separate small tests that each do only one thing?

An alternative would be to parametrize? (maybe a parametrized fixture?)

Reply:
You see I missed that day of the book club... ;) I'll split them up. (Maybe using parametrize but I think that might get messy here and in that case I'll just split them instead).